Rename user-defined actions to jobs (#3942)
timescale/docs: api/alter_job.md

WHERE proc_name = 'policy_compression' AND hypertable_name = 'conditions'
```

Reschedules continuous aggregate $JOB ID `1000` so that it next runs at 9:00:00 on 15 March, 2020:

```sql
SELECT alter_job(1000, next_start => '2020-03-15 09:00:00.0+00');
```

### Calculation of next start on failure

When a $JOB run results in a runtime failure, the next start of the $JOB is calculated taking into account both its `retry_period` and `schedule_interval`.
The `next_start` time is calculated using the following formula:
```
next_start = finish_time + consecutive_failures * retry_period ± jitter
```
where jitter (± 13%) is added to avoid the "thundering herds" effect.

<Highlight type="note">

To ensure that the `next_start` time is not put off indefinitely or produce timestamps so large they end up out of range, it is capped at 5*`schedule_interval`.
Also, more than 20 consecutive failures are not considered, so if the number of consecutive failures is higher, then it multiplies by 20.

> Additionally, in the case of $JOB with fixed schedules, the system ensures that if the next start calculated as specified, surpasses the next scheduled execution, then the $JOB is executed again at the next scheduled slot and not after that. This ensures that the $JOB does not miss scheduled executions.

Reviewer: ```suggestion
Additionally, for $JOBs with fixed schedules, the system ensures that if the next start ( calculated as specified), surpasses the next scheduled execution, the $JOB is executed again at the next scheduled slot and not after that. This ensures that the $JOB does not miss scheduled executions.
```

> Finally, there is a distinction between runtime failures that do not cause the $JOB to crash and $JOB crashes.

Reviewer: ```suggestion
There is a distinction between runtime failures that do not cause the $JOB to crash and $JOB crashes.
```

> In the event of a $JOB crash, the next start calculation still follows the above formula,

Reviewer: ```suggestion
In the event of a $JOB crash, the next start calculation follows the same formula,
```
